### PR TITLE
Reduce synchronous access to WorkingBeatmap properties

### PR DESCRIPTION
**Pull request by peppy**
I've added `Get*Async()` methods in a separate branch, but before resorting to using these, let's try and move as much access to the properties to conditional (using `*Loaded` as a precondition) or the `BackgroundDependencyLoader`/`LoadComponentAsync` pattern.

Note that editor components (`TimelinePart` derivatives) are not fixed yet. It will require a bit of restructuring to fix these components.

### CODE REVIEW

File: osu.Game/Screens/Select/BeatmapInfoWedge.cs [hunk 1]
```diff
@@ -65,56 +65,62 @@ protected override void PopOut()
...
-            float newDepth = lastContainer?.Depth + 1 ?? 0;
+            LoadComponentAsync(new BufferedWedgeInfo(beatmap)
+            {
+                Shear = -Shear,
+                Depth = info?.Depth + 1 ?? 0,
+            }, newInfo =>
+            {
+                // ensure we ourselves are visible if not already.
+                if (!IsPresent)
+                    this.FadeIn(250);
```
Comment (smoogipoo) [inline osu.Game/Screens/Select/BeatmapInfoWedge.cs]: Wait what, when does BufferedWedgeInfo even get an alpha of 0? It doesn't seem to ever be?

File: osu.Game/Screens/Select/BeatmapInfoWedge.cs [hunk 2]
```diff
@@ -65,56 +65,62 @@ protected override void PopOut()
...
-            var lastContainer = beatmapInfoContainer;
-            float newDepth = lastContainer?.Depth + 1 ?? 0;
+            LoadComponentAsync(new BufferedWedgeInfo(beatmap)
+            {
+                Shear = -Shear,
+                Depth = info?.Depth + 1 ?? 0,
+            }, newInfo =>
+            {
+                // ensure we ourselves are visible if not already.
+                if (!IsPresent)
```
Comment (smoogipoo) [inline osu.Game/Screens/Select/BeatmapInfoWedge.cs]: This is a weird condition, if the BeatmapInfoWedge is not present, then I see no way that the BufferedWedgeInfo will ever get displayed. The previous code did not have this condition.

**Review by smoogipoo — CHANGES_REQUESTED** [review]
Looks good apart from a few minor issues.

File: osu.Game/Screens/Select/BeatmapInfoWedge.cs (hunk 2 above)
Comment (peppy) [reply osu.Game/Screens/Select/BeatmapInfoWedge.cs]: is/was relying on AlwaysPresent above

File: osu.Game/Screens/Select/BeatmapInfoWedge.cs (hunk 1 above)
Comment (peppy) [reply osu.Game/Screens/Select/BeatmapInfoWedge.cs]: `BeatmapInfoWedge`, not `BufferedWedgeInfo`. `this.FadeIn`.

Comment (smoogipoo) [reply osu.Game/Screens/Select/BeatmapInfoWedge.cs]: Oops I misread.

**Review by smoogipoo — APPROVED** [review]

**Pull request merged**